frontend/public/components/cluster-settings/cluster-version.tsx

```diff
@@ -30,19 +45,36 @@ const ClusterVersionDetails: React.FC<ClusterVersionDetailsProps> = ({ obj }) =>
   );
 };
 
-export const ClusterVersionDetailsPage: React.FC<ClusterVersionDetailsPageProps> = (props) => (
-  <DetailsPage
-    {...props}
-    kind={clusterVersionReference}
-    pages={[navFactory.details(ClusterVersionDetails), navFactory.editYaml()]}
-    breadcrumbsFor={() => breadcrumbsForGlobalConfig(ClusterVersionModel.label, props.match.url)}
-  />
-);
+export const ClusterVersionDetailsPage: React.FC<ClusterVersionDetailsPageProps> = (props) => {
+  const clusterVersionResource = {
+    kind: props.kind,
+    kindObj: props.kindObj,
+    name: props.name,
+    isList: false,
+  };
+  const [clusterVersion] = useK8sWatchResource<ClusterVersionKind>(clusterVersionResource);
+  const canUpgrade = useCanClusterUpgrade(clusterVersion);
+
+  return (
+    <DetailsPage
+      {...props}
+      kind={clusterVersionReference}
+      pages={[
+        navFactory.details(ClusterVersionDetails),
+        navFactory.editYaml(canUpgrade ? editYamlComponent : viewYamlComponent),
```

Comment:
@spadgett, is there a better way to conditionally set the YAML editor to read only?